### ISSUE TITLE
 ld.lld: error: undefined symbol: ovr_ApplicationOptions_SetDestinationApiName while packaging using UE5.3

### ISSUE THREAD

**Issue by LeeHammoud** [oculus-samples/Unreal-SharedSpaces#78]
Hey,

I have the latest Meta's fork of Unreal 5.3, along with the latest plugin version of the MetaXR Platforms & OculusUtils plugin from this repo placed in a project. I made sure engine version association of the plugins are removed instead as well to work with meta's fork. Everything seems to be enabled properly, and in editor- it works fine. However, upon packaging- I get linkage errors

````
UATHelper: Packaging (Android (ASTC)): ld.lld: error: undefined symbol: ovr_ApplicationOptions_SetDestinationApiName
UATHelper: Packaging (Android (ASTC)): >>> referenced by OVRPlatformOptionsConverters.h:95 (.../Plugins/MetaXRPlatform/Source/OVRPlatform/Private/OVRPlatformOptionsConverters.h:95)
UATHelper: Packaging (Android (ASTC)): >>>               OVRPlatform/Module.OVRPlatform.3.cpp.o:(FOvrApplicationOptionsConverter::FOvrApplicationOptionsConverter(FOvrApplicationOptions const&))
UATHelper: Packaging (Android (ASTC)): >>> referenced by OVRPlatformOptionsConverters.h:95 (.../Plugins/MetaXRPlatform/Source/OVRPlatform/Private/OVRPlatformOptionsConverters.h:95)
UATHelper: Packaging (Android (ASTC)): >>>               OVRPlatform/Module.OVRPlatform.3.cpp.o:(FOvrApplicationOptionsConverter::FOvrApplicationOptionsConverter(FOvrApplicationOptions const&))
UATHelper: Packaging (Android (ASTC)): ld.lld: error: undefined symbol: ovr_ApplicationOptions_SetLobbySessionId
UATHelper: Packaging (Android (ASTC)): >>> referenced by OVRPlatformOptionsConverters.h:96 (.../Plugins/MetaXRPlatform/Source/OVRPlatform/Private/OVRPlatformOptionsConverters.h:96)
UATHelper: Packaging (Android (ASTC)): >>>               OVRPlatform/Module.OVRPlatform.3.cpp.o:(FOvrApplicationOptionsConverter::FOvrApplicationOptionsConverter(FOvrApplicationOptions const&))
UATHelper: Packaging (Android (ASTC)): >>> referenced by OVRPlatformOptionsConverters.h:96 (.../Plugins/MetaXRPlatform/Source/OVRPlatform/Private/OVRPlatformOptionsConverters.h:96)
UATHelper: Packaging (Android (ASTC)): >>>               OVRPlatform/Module.OVRPlatform.3.cpp.o:(FOvrApplicationOptionsConverter::FOvrApplicationOptionsConverter(FOvrApplicationOptions const&))
UATHelper: Packaging (Android (ASTC)): ld.lld: error: undefined symbol: ovr_ApplicationOptions_SetMatchSessionId
UATHelper: Packaging (Android (ASTC)): >>> referenced by OVRPlatformOptionsConverters.h:97 (.../Plugins/MetaXRPlatform/Source/OVRPlatform/Private/OVRPlatformOptionsConverters.h:97)
UATHelper: Packaging (Android (ASTC)): >>>               OVRPlatform/Module.OVRPlatform.3.cpp.o:(FOvrApplicationOptionsConverter::FOvrApplicationOptionsConverter(FOvrApplicationOptions const&))
UATHelper: Packaging (Android (ASTC)): >>> referenced by OVRPlatformOptionsConverters.h:97 (.../Plugins/MetaXRPlatform/Source/OVRPlatform/Private/OVRPlatformOptionsConverters.h:97)
UATHelper: Packaging (Android (ASTC)): >>>               OVRPlatform/Module.OVRPlatform.3.cpp.o:(FOvrApplicationOptionsConverter::FOvrApplicationOptionsConverter(FOvrApplicationOptions const&))
UATHelper: Packaging (Android (ASTC)): ld.lld: error: undefined symbol: ovr_ApplicationOptions_SetRoomId
UATHelper: Packaging (Android (ASTC)): >>> referenced by OVRPlatformOptionsConverters.h:98 (.../Plugins/MetaXRPlatform/Source/OVRPlatform/Private/OVRPlatformOptionsConverters.h:98)
UATHelper: Packaging (Android (ASTC)): >>>               OVRPlatform/Module.OVRPlatform.3.cpp.o:(FOvrApplicationOptionsConverter::FOvrApplicationOptionsConverter(FOvrApplicationOptions const&))
UATHelper: Packaging (Android (ASTC)): ld.lld: error: undefined symbol: ovr_AppDownloadProgressResult_GetDownloadBytes
UATHelper: Packaging (Android (ASTC)): >>> referenced by OVRPlatformModels.cpp:246 (.../Plugins/MetaXRPlatform/Source/OVRPlatform/Private/OVRPlatformModels.cpp:246)
UATHelper: Packaging (Android (ASTC)): >>>               OVRPlatform/Module.OVRPlatform.3.cpp.o:(FOvrAppDownloadProgressResult::Update(ovrAppDownloadProgressResult*, std::__ndk1::shared_ptr<TOvrMessageHandle>))
UATHelper: Packaging (Android (ASTC)): ld.lld: error: undefined symbol: ovr_AppDownloadProgressResult_GetDownloadedBytes
UATHelper: Packaging (Android (ASTC)): >>> referenced by OVRPlatformModels.cpp:247 (.../Plugins/MetaXRPlatform/Source/OVRPlatform/Private/OVRPlatformModels.cpp:247)
UATHelper: Packaging (Android (ASTC)): >>>               OVRPlatform/Module.OVRPlatform.3.cpp.o:(FOvrAppDownloadProgressResult::Update(ovrAppDownloadProgressResult*, std::__ndk1::shared_ptr<TOvrMessageHandle>))
UATHelper: Packaging (Android (ASTC)): ld.lld: error: undefined symbol: ovr_AppDownloadProgressResult_GetStatusCode
UATHelper: Packaging (Android (ASTC)): >>> referenced by OVRPlatformModels.cpp:248 (.../Plugins/MetaXRPlatform/Source/OVRPlatform/Private/OVRPlatformModels.cpp:248)
UATHelper: Packaging (Android (ASTC)): >>>               OVRPlatform/Module.OVRPlatform.3.cpp.o:(FOvrAppDownloadProgressResult::Update(ovrAppDownloadProgressResult*, std::__ndk1::shared_ptr<TOvrMessageHandle>))
UATHelper: Packaging (Android (ASTC)): ld.lld: error: undefined symbol: ovr_AppDownloadResult_GetTimestamp
UATHelper: Packaging (Android (ASTC)): >>> referenced by OVRPlatformModels.cpp:271 (.../Plugins/MetaXRPlatform/Source/OVRPlatform/Private/OVRPlatformModels.cpp:271)
UATHelper: Packaging (Android (ASTC)): >>>               OVRPlatform/Module.OVRPlatform.3.cpp.o:(FOvrAppDownloadResult::Update(ovrAppDownloadResult*, std::__ndk1::shared_ptr<TOvrMessageHandle>))
UATHelper: Packaging (Android (ASTC)): ld.lld: error: undefined symbol: ovr_ApplicationVersion_GetReleaseDate
UATHelper: Packaging (Android (ASTC)): >>> referenced by OVRPlatformModels.cpp:303 (.../Plugins/MetaXRPlatform/Source/OVRPlatform/Private/OVRPlatformModels.cpp:303)
UATHelper: Packaging (Android (ASTC)): >>>               OVRPlatform/Module.OVRPlatform.3.cpp.o:(FOvrApplicationVersion::Update(ovrApplicationVersion*, std::__ndk1::shared_ptr<TOvrMessageHandle>))
UATHelper: Packaging (Android (ASTC)): ld.lld: error: undefined symbol: ovr_ApplicationVersion_GetSize
UATHelper: Packaging (Android (ASTC)): >>> referenced by OVRPlatformModels.cpp:304 (.../Plugins/MetaXRPlatform/Source/OVRPlatform/Private/OVRPlatformModels.cpp:304)
UATHelper: Packaging (Android (ASTC)): >>>               OVRPlatform/Module.OVRPlatform.3.cpp.o:(FOvrApplicationVersion::Update(ovrApplicationVersion*, std::__ndk1::shared_ptr<TOvrMessageHandle>))
UATHelper: Packaging (Android (ASTC)): ld.lld: error: undefined symbol: ovr_AvatarEditorResult_GetRequestSent
UATHelper: Packaging (Android (ASTC)): >>> referenced by OVRPlatformModels.cpp:441 (.../Plugins/MetaXRPlatform/Source/OVRPlatform/Private/OVRPlatformModels.cpp:441)
UATHelper: Packaging (Android (ASTC)): >>>               OVRPlatform/Module.OVRPlatform.3.cpp.o:(FOvrAvatarEditorResult::Update(ovrAvatarEditorResult*, std::__ndk1::shared_ptr<TOvrMessageHandle>))
UATHelper: Packaging (Android (ASTC)): ld.lld: error: undefined symbol: ovr_Destination_GetShareableUri
UATHelper: Packaging (Android (ASTC)): >>> referenced by OVRPlatformModels.cpp:538 (.../Plugins/MetaXRPlatform/Source/OVRPlatform/Private/OVRPlatformModels.cpp:538)
UATHelper: Packaging (Android (ASTC)): >>>               OVRPlatform/Module.OVRPlatform.3.cpp.o:(FOvrDestination::Update(ovrDestination*, std::__ndk1::shared_ptr<TOvrMessageHandle>))
UATHelper: Packaging (Android (ASTC)): ld.lld: error: undefined symbol: ovr_Purchase_GetDeveloperPayload
UATHelper: Packaging (Android (ASTC)): >>> referenced by OVRPlatformModels.cpp:1609 (.../Plugins/MetaXRPlatform/Source/OVRPlatform/Private/OVRPlatformModels.cpp:1609)
UATHelper: Packaging (Android (ASTC)): >>>               OVRPlatform/Module.OVRPlatform.3.cpp.o:(FOvrPurchase::Update(ovrPurchase*, std::__ndk1::shared_ptr<TOvrMessageHandle>))
UATHelper: Packaging (Android (ASTC)): ld.lld: error: undefined symbol: ovr_Purchase_GetReportingId
UATHelper: Packaging (Android (ASTC)): >>> referenced by OVRPlatformModels.cpp:1613 (.../Plugins/MetaXRPlatform/Source/OVRPlatform/Private/OVRPlatformModels.cpp:1613)
UATHelper: Packaging (Android (ASTC)): >>>               OVRPlatform/Module.OVRPlatform.3.cpp.o:(FOvrPurchase::Update(ovrPurchase*, std::__ndk1::shared_ptr<TOvrMessageHandle>))
UATHelper: Packaging (Android (ASTC)): ld.lld: error: undefined symbol: ovr_UserAccountAgeCategory_Get
````
and ends with

````
PackagingResults: Error: undefined symbol: ovr_ApplicationOptions_SetDestinationApiName
PackagingResults: Error: undefined symbol: ovr_ApplicationOptions_SetLobbySessionId
PackagingResults: Error: undefined symbol: ovr_ApplicationOptions_SetMatchSessionId
PackagingResults: Error: undefined symbol: ovr_ApplicationOptions_SetRoomId
PackagingResults: Error: undefined symbol: ovr_AppDownloadProgressResult_GetDownloadBytes
PackagingResults: Error: undefined symbol: ovr_AppDownloadProgressResult_GetDownloadedBytes
PackagingResults: Error: undefined symbol: ovr_AppDownloadProgressResult_GetStatusCode
PackagingResults: Error: undefined symbol: ovr_AppDownloadResult_GetTimestamp
PackagingResults: Error: undefined symbol: ovr_ApplicationVersion_GetReleaseDate
PackagingResults: Error: undefined symbol: ovr_ApplicationVersion_GetSize
PackagingResults: Error: undefined symbol: ovr_AvatarEditorResult_GetRequestSent
PackagingResults: Error: undefined symbol: ovr_Destination_GetShareableUri
PackagingResults: Error: undefined symbol: ovr_Purchase_GetDeveloperPayload
PackagingResults: Error: undefined symbol: ovr_Purchase_GetReportingId
PackagingResults: Error: undefined symbol: ovr_UserAccountAgeCategory_GetAgeCategory
PackagingResults: Error: undefined symbol: ovr_AbuseReport_ReportRequestHandled
PackagingResults: Error: undefined symbol: ovr_Application_CancelAppDownload
PackagingResults: Error: undefined symbol: ovr_Message_GetAppDownloadResult
PackagingResults: Error: undefined symbol: ovr_Application_CheckAppDownloadProgress
PackagingResults: Error: undefined symbol: ovr_Message_GetAppDownloadProgressResult
PackagingResults: Error: too many errors emitted, stopping now (use -error-limit=0 to see all errors)
PackagingResults: Error: linker command failed with exit code 1 (use -v to see invocation)
PackagingResults: Error: Unknown Error
````

Then of course, the project fails to package. Any idea what's going on?

**Comment by LeeHammoud** [comment]
Turning off the Oculus Online Subsystem stops the packaging from getting this error. Quite weird to get these linkage errors because of that, although I see that MetaXR platforms does replace OOS after all.

[EDITED]

Buuut you need OOS for MetaXRPlatform to initialize and work...

**Comment by LeeHammoud** [comment]
Buuut you need OOS for MetaXRPlatform to work... This is problematic...

**Comment by alefriant** [comment]
I have the same issue here, it's impossible to package with the new version of UE532 and seems to be the OVR the problem. 
Can you run it in pc mode? I mean with the Play in the editor mode? During the last week of February I lost that feature too + a lot of bugs during that week.

**Comment by LeeHammoud** [comment]
Oh no, I'm stupid. I, instead of doing the obvious- that is downloading the SDK from https://developer.oculus.com/downloads/package/unreal-5-platform-sdk-plugin/, I copied the plugin from the SharedSpaces which is causing linkage errors. My bad for anyone who got confused.

Cheers!

@alefriant

**Comment by alefriant** [comment]
Up!

**Comment by jasonmeisel** [comment]
This should be solved by the latest change (197f26ea6845b5f49dad6f840e909b280a6b0db5). If this resolves the issue for you, please let us know and close the issue.

**Comment by jasonmeisel** [comment]
Note that you do not need OnlineSubsystemOculus to use the Meta XR Platform plugin. In fact, it will not work if you enable it. See [here](https://github.com/oculus-samples/Unreal-SharedSpaces/blob/main-5.x/Documentation/SharedSpaces.md#d-oculus-application-configuration).
